### ISSUE TITLE
Blank screen on Chrome Android

### ISSUE THREAD

**Issue by georginio** [jeeliz/jeelizFaceFilter#157]
Sorry, I didn't response to issue #152

This are screens from the webglreport

Device is Android 6.0, Chrome 76

![Screenshot_20200210-141720](https://user-images.githubusercontent.com/8328003/74142862-2a8a0780-4c13-11ea-82a1-ecf217aeb9ce.png)
![Screenshot_20200210-141731](https://user-images.githubusercontent.com/8328003/74142869-2f4ebb80-4c13-11ea-8f40-7ab3ea9b0892.png)
![Screenshot_20200210-141813](https://user-images.githubusercontent.com/8328003/74142875-337ad900-4c13-11ea-9540-d0a8a1848ab5.png)

Also full console log from jeeliz anonymous demo

![Screenshot from 2020-02-10 14-37-34](https://user-images.githubusercontent.com/8328003/74142921-4beaf380-4c13-11ea-95cb-6eed29563ceb.png)

**Comment by xavierjs** [comment]
Hi,

Thank you for this detailed bug report.
Your device does not seem to be compatible.
To run a deep learning neural network, we need to be able to do render to texture with 16 or 
32 bits precision.
Even if your OS and browser are up to date (like in your case), if the hardware cannot we can do nothing..
.
With WebGL2 it is guaranteed that we can do render to texture with 16 bits precision but WebGL2 is disabled in your device (I think it is because your hardware cannot run WebGL2).
With WebGL1 we need some extensions, and to be able to do Render to texture with 16 or 32 bits precision, which is not always implemented on old hardware.
Your GPU is an Adreno 330. It seems a bit outdated. If it was not, WebGL2 would be enabled.

**Comment by oliverhausler** [comment]
@xavierjs Just an idea. Wouldn't it be technically possible to fallback to black and white on such devices? I don't know if WEBGL1 could do 8 bits, but 8 bits should be more than enough for black and white. If this can be easily done, users with older devices would at least see something instead nothing.

**Comment by xavierjs** [comment]
It would be possible, but too slow I think.
In each shader you whould have to pack & unpack the values into RGBA 8bits channels.
It will add processing time. And also currently we use the RGBA channel to run 4 different neural networks in parallel to have a better stabilization. With this system we would be only able to run 1 neural network at the same time.

We currently don't target devices which are too old. If the GPU is currently not able to do RTT with more than 8 bits precision, it will also be to slow to run the neural network models correctly, because everything is sized for an Iphone6 / Motorolla XPlay / Sony Xperia Zperformance as minimum device.

I think for low end devices an image based, server-side fallback makes sense. It would also be useful for users who don't want to share their camera or if an other application is already using the camera.

**Comment by oliverhausler** [comment]
Thanks for the assessment. I was hoping it was a good idea, but if it's not feasible we can't do much.

This is another issue, but regarding what you are saying about the neural networks: I am not seeing this as the critical factor for quality. The results from jeeliz/neural networks is rather good as I find. The main problem I see is with the 3D models themselves. For example, the Statue of Liberty is well done. When you turn your head, the model scales appropriately, but most examples seem to not properly track the head because the 3D model/view angles/size calculations for the mask are wrong, not what comes as data from jeeliz. I haven't digged in so deep, but from what I visually see, my impression is that quality varies a lot based on programming of the mask.

**Comment by xavierjs** [comment]
Sure, some demos are a bit deprecated, they were done at the beginning of the project and the pivot point is badly positionned. There was a bug in the Three.js helper so in some demos the Z of the pivot point/mask is displaced to compensate the bug. Since this bug is solved some demos may be a bit screwed. If you tested a demo which looks particularly bad, you are welcome to post its URL in the issues section and I will take a look.

**Comment by oliverhausler** [comment]
Cool, I will create a new issue for this.
